packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -1605,3 +1613,668 @@ class _RestorableScrollOffset extends RestorableValue<double?> {
+                instance
+                  ..onDown = _handleDragDown
+                  ..onStart = _handleDragStart
+                  ..onUpdate = _handleDragUpdate
+                  ..onEnd = _handleDragEnd
+                  ..onCancel = _handleDragCancel
+                  ..minFlingDistance = _physics?.minFlingDistance
+                  ..minFlingVelocity = _physics?.minFlingVelocity
+                  ..maxFlingVelocity = _physics?.maxFlingVelocity
+                  ..velocityTrackerBuilder = _configuration.velocityTrackerBuilder(context)
+                  ..dragStartBehavior = widget.dragStartBehavior
+                  ..gestureSettings = _mediaQueryGestureSettings;
+              },
+            ),
+          };
+          // Cancel the active hold/drag (if any) because the gesture recognizers
+          // will soon be disposed by our RawGestureDetector, and we won't be
+          // receiving pointer up events to cancel the hold/drag.
+          _handleDragCancel();
+          _lastCanDrag = value;
+          _lastAxisDirection = widget.axis;
+          if (_gestureDetectorKey.currentState != null) {
+            _gestureDetectorKey.currentState!.replaceGestureRecognizers(_gestureRecognizers);
+          }
+        }
+        return;
+    }
+    // If we aren't scrolling diagonally, the default drag gesture recognizer
+    // is used.
+    super.setCanDrag(value);
+  }
+
+  @override
+  Widget _buildChrome(BuildContext context, Widget child) {
+    final ScrollableDetails details = ScrollableDetails(
+      direction: widget.axisDirection,
+      controller: _effectiveScrollController,
+      clipBehavior: widget.clipBehavior,
+    );
+    // Skip building a scrollbar here, the dual scrollbar is added in
+    // TwoDimensionalScrollableState.
+    return _configuration.buildOverscrollIndicator(context, child, details);
+  }
+}
+
+/// Specifies how to configure the [DragGestureRecognizer]s of a
+/// [TwoDimensionalScrollable].
+enum DiagonalDragBehavior {
+  /// This behavior will not allow for any diagonal scrolling.
+  ///
+  /// Drag gestures in one direction or the other will lock the input axis until
+  /// the gesture is released.
+  none,
+
+  /// This behavior will only allow diagonal scrolling on a weighted
+  /// scale per gesture event.
+  ///
+  /// This means that after initially evaluating the drag gesture, the weighted
+  /// evaluation (based on [kTouchSlop]) stands until the gesture is released.
+  weightedEvent,
+
+  /// This behavior will only allow diagonal scrolling on a weighted
+  /// scale that is evaluated throughout a gesture event.
+  ///
+  /// This means that during each update to the drag gesture, the scrolling
+  /// axis will be allowed to scroll diagonally if it exceeds the
+  /// [kTouchSlop].
+  weightedContinuous,
+
+  /// This behavior allows free movement in any and all directions when
+  /// dragging.
+  free,
+}
+
+/// A widget that scrolls in two dimensions.
```

Comment:
compare this with the doc on TwoDimensionalScrollView:

> A widget that scrolls in both dimensions.

Any chance the summary line for both could differentiate a little more? Right now it sounds like they basically do the same?

Reply:
You found my side quest. 🏆 

Since I copied most of the semantics from 1D scrolling docs, this is helping me find places where we are lacking in our documentation for one dimensional scrolling. :)

Fixed for 1D and 2D now.